webapp/src/components/onboarding/LLMConfigurationForm.tsx

```diff
@@ -0,0 +1,564 @@
+						EmbeddingModelRequiredFields.map(field => (
+							<div key={field.name} className='mt-2'>
+								<InputField<LLMConfigurationFormValues>
+									name={field.name as keyof LLMConfigurationFormValues}
+									rules={{ required: embeddingModel?.value ? `${field.label} is required` : false }}
+									label={field.label}
+									type='text'
+									control={control}
+									placeholder={field.placeholder}
+									disabled={!LLMModel || !LLMModel.value}
+								/>
+							</div>
+						))}
+				</div>
+			</div>
+
+			{isTablet && (
+				<div className='flex gap-8 mt-4'>
+					<div className='flex-1'>
+						{LLMModelRequiredFields.length > 0 &&
+							LLMModelRequiredFields.map(field => (
+								<div key={field.name}>
+									<InputField<LLMConfigurationFormValues>
+										name={field.name as keyof LLMConfigurationFormValues}
+										rules={{ required: LLMModel?.value ? `${field.label} is required` : false }}
+										label={field.label}
+										type='text'
+										control={control}
+										placeholder={field.placeholder}
+										disabled={!LLMModel || !LLMModel.value}
+									/>
+								</div>
+							))}
+					</div>
+					<div className='flex-1'>
+						{EmbeddingModelRequiredFields.length > 0 &&
+							EmbeddingModelRequiredFields.map(field => (
+								<div key={field.name}>
+									<InputField<LLMConfigurationFormValues>
+										name={field.name as keyof LLMConfigurationFormValues}
+										rules={{
+											required: embeddingModel?.value ? `${field.label} is required` : false
+										}}
+										label={field.label}
+										type='text'
+										control={control}
+										placeholder={field.placeholder}
+										// value={embeddingAPIKeyValue}
+										disabled={!embeddingModel || !embeddingModel.value}
+									/>
+								</div>
+							))}
+					</div>
+				</div>
+			)}
+
+			<hr className='mt-14 mb-5' />
+			<div className='flex'>
+				<button
+					className='w-[137px] h-[41px] border border-gray-200 rounded-lg text-sm'
+					type='button'
+					onClick={updateOnboardedStatus}
+				>
+					I&apos;ll do this later
+				</button>
+				<button
+					className='ml-auto w-[140px] h-[41px] disabled:bg-primary-200 bg-primary-500 text-white rounded-lg flex justify-center items-center text-sm'
+					type='submit'
+					disabled={!LLMModel?.value && !embeddingModel?.value}
+				>
+					{submitting ? (
+						<ButtonSpinner className='mt-1 me-2' />
+					) : (
+						<>
+							<span className='text-sm'>Get Started</span>
+							<ChevronRightIcon className='ml-2 h-5 w-5' />
+						</>
+					)}
+				</button>
+			</div>
+		</form>
+	);
+};
+
+export default LLMConfigurationForm;
+
+interface Team {
+	_id: string;
+	ownerId: string;
+	name: string;
+	orgId: string;
+	members: string[];
+	dateCreated: string;
+	permissions: Record<string, string>;
+	llmModel: Model;
+	embeddingModel: Model;
+}
+
+interface Model {
+	_id: string;
```

Comment:
Are these interfaces necessary? If they can be moved to struct/ or removed that would be preferred.

Reply:
i'd like to keep these interfaces to make refactors much easier if required.
i've moved the  interface to `types/teammodels.ts` . I also introduced a type safety to `getTeamModels`
by adding `export function getTeamModels(body, dispatch: GetTeamModelsDispatch, ...`